File: android/app/src/main/java/com/keksovmen/flowerbox/entities/Property.kt
package com.keksovmen.flowerbox.entities

import androidx.room.Entity
import androidx.room.ForeignKey

@Entity(
    primaryKeys = ["parent_id", "id"],
    foreignKeys = [ForeignKey(
        entity = FlowerBox::class,
        parentColumns = ["unique_id"],
        childColumns = ["parent_id"],
        onDelete = ForeignKey.CASCADE
    )]
)
data class Property(
    val name: String,
    val description: String,
    val value_type: String,
    var parent_id: Int,
    val id: Int,
    val tid: Int,
    val min_value: Float,
    val max_value: Float,
    var value: Float,
)

File: emulator/property.py
class Property:
	def __init__(self, name, description, value_type, min_value, max_value, tid, id, value) -> None:
		self.name = name
		self.description = description
		self.value_type = value_type
		self.min_value = min_value
		self.max_value = max_value
		self.tid = tid
		self.id = id
		self.value = value